module/directory.go

```diff
@@ -33,10 +33,16 @@ func NewDirectory() *Directory {
 
 // Sync is a lazy implementation to refresh all of the module dependencies to the closest semver.
 func (d *Directory) Sync() {
+	names := make(map[string]bool, 0)
```

Comment:
as a nit, for these maps which are only use to check if a key has been "seen", it's generally better practice to have it be `map[string]struct{}` as then you don't need to worry about what the value is, just whether or not the key exists.